src/main/scala/io/github/edefritz/commands/Tile38Command.scala

```diff
@@ -2,17 +2,152 @@ package io.github.edefritz.commands
 
 import io.github.edefritz.commands.GetCommand._
 import io.github.edefritz.commands.OutputCommand.OutputCommandArgument
+import io.github.edefritz.commands.SetCommand.{
+  Ex,
+  Fields,
+  SetCommandArgument,
+  SetCommandFields,
+  SetCommandInputFormat,
+  SetCondition,
+  SetString,
+  Bounds => SetBounds,
+  Hash => SetHash,
+  Object => SetObject,
+  Point => SetPoint
+}
 import io.lettuce.core.codec.StringCodec
-import io.lettuce.core.protocol.{CommandArgs, CommandType, ProtocolKeyword}
+import io.lettuce.core.protocol.{ CommandArgs, CommandType, ProtocolKeyword }
 
 import scala.jdk.CollectionConverters._
-import scala.util.{Success, Try}
+import scala.util.{ Success, Try }
 
 sealed trait Tile38Command {
   val protocolKeyword: ProtocolKeyword
   def compileArguments(): Try[CommandArgs[String, String]]
 }
 
+final case class SetCommand(
+    key: String,
+    id: String,
+    // TODO: How to not make this as an Option so users don't have to pass Some() ???
+    inputFormat: SetCommandArgument with SetCommandInputFormat,
+    fields: Option[Fields],
+    ex: Option[Ex],
+    setCondition: Option[SetCommandArgument with SetCondition]
+) extends Tile38Command {
+  override val protocolKeyword: ProtocolKeyword = CommandType.SET
+
+  override def compileArguments(): Try[CommandArgs[String, String]] = {
+    val args = new CommandArgs(StringCodec.UTF8)
+    args.add(key)
+    args.add(id)
+
+    fields.foreach { f =>
+      f.fields.foreach {
+        case (fieldName, fieldValue) => args.add(f.keyword).add(fieldName).add(fieldValue)
+      }
+    }
+
+    ex.foreach { e =>
+      args.add(e.keyword).add(e.seconds)
+    }
+
+    setCondition.foreach { n =>
+      args.add(n.keyword)
+    }
+
+    inputFormat match {
+      case SetPoint(lat, lon, z) =>
+        args
+          .add(inputFormat.keyword)
+          .add(lat)
+          .add(lon)
+        if (z.isDefined) args.add(z.get)
+      case SetObject(geojson) => args.add(inputFormat.keyword).add(geojson)
+      case SetBounds(minlat, minlon, maxlat, maxlon) =>
+        args.add(inputFormat.keyword).add(minlat).add(minlon).add(maxlat).add(maxlon)
```

Comment:
To make sure we don't forget to use any of these arguments (e.g. copy paste issue), we could add strict compiler settings (https://gist.github.com/tabdulradi/aa7450921756cd22db6d278100b2dac8) so the complile will fail compilation if any of those params is not used.